api/backend/blueflood/blueflood.go

```diff
@@ -218,18 +260,21 @@ var samplerMap map[api.SampleMethod]struct {
 
 // Blueflood keys the resolution param to a java enum, so we have to convert
 // between them.
-func bluefloodResolution(r int64) resolution {
+func (c Config) bluefloodResolution(resolution int64, timestamp int64) string {
```

Comment:
i feel like this logic of:
```
given a desired resolution, timerange (we pass in current time), and current time, give us a resolution
```
can be refactored a bit more. specifically, i feel like we're going to use it when we implement graphite backend (or other different backend).